src/components/CurrentUser/CurrentUser.jsx

```diff
@@ -1,12 +1,92 @@
 import React from 'react';
 import './CurrentUser.scss';
+import PropTypes from 'prop-types';
+import { getUser } from '../../api';
 
-export const CurrentUser = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+export class CurrentUser extends React.PureComponent {
+  state = {
+    user: {},
+    loadError: false,
+  }
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+  static propTypes = {
+    userId: PropTypes.number.isRequired,
+    clearUser: PropTypes.func.isRequired,
+  }
+
+  componentDidMount = () => {
+    this.updateUserInfo(this.props.userId);
+  }
+
+  componentDidUpdate = ({ userId }) => {
+    const newUserId = this.props.userId;
+
+    if (userId !== newUserId) {
+      this.updateUserInfo(newUserId);
+    }
+  }
+
+  updateUserInfo = async(userId) => {
+    try {
+      await getUser(userId).then((user) => {
+        if (user.data === null) {
+          throw new Error('User info missing');
+        }
+
+        this.setState({
+          user: user.data,
+          loadError: false,
+        });
+      });
+    } catch (error) {
+      this.setState({
+        user: {},
+        loadError: true,
+      });
+    }
+  }
+
+  render() {
+    const { user, loadError } = this.state;
+    const { clearUser } = this.props;
+
+    return (
+      <div className="CurrentUser">
+        <h2 className="CurrentUser__title">
+          <span>
+            Selected user:&nbsp;
+            {user && user.id}
```

Comment:
we can check if user exists outside return statement and return null if it doesn't. In this case you won't need to do 4 user checks in return.